### ISSUE TITLE
SwinjectAutoregistration: ambiguous reference when initializer takes 0 parameters

### ISSUE THREAD

**Issue by Rivukis** [Swinject/SwinjectAutoregistration#34]
Firstly, I really like Swinject and SwinjectAutoregistration. Nice work!

This might be a bug in the Swift compiler. Using Xcode 9.2, Swift complains of an ambiguous reference when initializer takes zero parameters. Not sure how you can fix this, though.

Example:
```
protocol Person { }
class PetOwner: Person {
    init() { }
}
let container = Container()
container.autoregister(Person.self, initializer: PetOwner.init) // Swift Error: Ambiguous use of 'autoregister(_:name:initializer:)'
```

**Comment by Rivukis** [comment]
This is a hack fix, but it worked for me. I'm explicitly passing in Void into the initializer.

The Void being in a tuple is to remove the Swift warning: 'When calling this function in Swift 4 or later, you must pass a '()' tuple; did you mean for the input type to be '()'?'. This is normally a good warning. However, to make the overloaded function work correctly, I needed to explicitly pass in Void.

```
@discardableResult
public func autoregister<Service>(_ service: Service.Type, name: String? = nil, initializer: @escaping ((Void)) -> Service) -> ServiceEntry<Service> {
   return self.register(service.self, name: name, factory: { res in 
    initializer(Void())
   } as (Resolver) -> Service)
}
```

**Comment by Rivukis** [comment]
After forking the repo to fix the issue, I noticed that the issue has already been fixed, and I was using 2.1.0. Sorry about all the commotion for an issue that's already been fixed.

I've made a pull request that adds a test for auto resolving a type that has an initializer with zero parameters. Feel free to merge it or reject it if you don't think it's necessary.
